plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/containers/N4JSAllContainersState.java

```diff
@@ -155,18 +160,25 @@ private void tryValidateManifest(final IResourceDelta delta) {
 				final ResourceSet resourceSet = core.createResourceSet(Optional.of(project));
 				final Resource resource = resourceSet.getResource(manifestLocation, true);
 				final Job job = Job.create("Update validation markers for " + resource.getURI(), monitor -> {
-					validatorExtension.updateValidationMarkers(manifest, resource, ALL, monitor);
-					return OK_STATUS;
+					ISchedulingRule rule = builder.getRule();
```

Comment:
@szarnekow we have noticed another race condition triggered from the builder. This seems to fix it, without affecting overall behaviour. Could you take a look too?

Reply:
(needed to merge already since the sprint ends now)

Reply:
I recommend to set the rule on the job itself